boltdb/bolt_store.go

```diff
@@ -3,413 +3,169 @@ package boltdb
 import (
 	"bytes"
 	"fmt"
-	"github.com/bmeg/arachne/aql"
-	"github.com/bmeg/arachne/gdbi"
+	"log"
+	//"github.com/bmeg/arachne/aql"
+	//"github.com/bmeg/arachne/gdbi"
+	"github.com/bmeg/arachne/kvgraph"
 	"github.com/boltdb/bolt"
-	proto "github.com/golang/protobuf/proto"
-	//"github.com/golang/protobuf/ptypes/struct"
-	//"log"
 )
 
-//Outgoing edges
-//key: src 0x00 dest 0x00 edgeid
-//value: edge properties
-var OEdgeBucket = []byte("oedges")
+var GRAPH_BUCKET = []byte("graph")
 
-//Incoming edges
-//key: dest 0x00 src 0x00 edgeid
-//value: blank
-var IEdgeBucket = []byte("iedges")
-
-//Vertices
-//key: vertex id
-//value: vertex properties
-var VertexBucket = []byte("vertices")
-
-type BoltArachne struct {
-	db *bolt.DB
-}
-
-func NewBoltArachne(path string) gdbi.ArachneInterface {
+func BoltBuilder(path string) (kvgraph.KVInterface, error) {
```

Comment:
`NewBoltKV` is the most idiomatic name

Reply:
Suggestion?

Reply:
Also, try not to return interfaces unless you really need to. It's often better to allow the caller access to the concrete type, in case there are type-specific methods.